addons/account/account_move_line.py

```diff
@@ -1359,7 +1359,7 @@ def create(self, cr, uid, vals, context=None, check=True):
             account_id = 'account_collected_id'
             base_sign = 'base_sign'
             tax_sign = 'tax_sign'
-            if journal.type in ('purchase_refund', 'sale_refund') or (journal.type in ('cash', 'bank') and total < 0):
+            if journal.type in ('purchase_refund', 'sale_refund') or (journal.type in ('cash', 'bank') and total < 0 and tax_id.type_tax_use != 'sale'):
```

Comment:
Hi Quentin,

Without this line, if taxes are set correctly, the use of a sale tax involves the selection of refund information on the tax (basically the Refund Base Code Sign
 and  Refund Tax Code Sign fields which are generally set to -1). This one will therefore caused an error in the sign of the base amount and the tax amount. 
More simply, if you want to introduce an income directly from a bank statement and if you have to define a sale taxe on this one, the sign of the amount of the base and the tax will be wrong without this line.